### ISSUE TITLE
Define metrics based on `TorchMetrics`

### ISSUE THREAD

**Issue by shackett** [napistu/napistu-torch#23]
```python
from torchmetrics import AUROC, AveragePrecision, Accuracy, MetricCollection

# Instead of your MetricRegistry
metrics = MetricCollection({
    'auc': AUROC(task='binary'),
    'ap': AveragePrecision(task='binary'),
    'acc': Accuracy(task='binary')
})

# Automatic accumulation across batches
for batch in dataloader:
    preds = model(batch)
    metrics.update(preds, targets)

results = metrics.compute()  # {'auc': 0.92, 'ap': 0.87, ...}
```

**Comment by shackett** [comment]
Closing as not-planned for now.
After evaluating torchmetrics for our current use case, we're sticking with sklearn-based metrics for the following reasons:

Full-batch evaluation: We evaluate on the entire val/test set in one forward pass, so torchmetrics' batch accumulation features aren't needed
Custom weighting complexity: Our relation-weighted AUC metric requires custom stratification, weight aggregation, and relation name decoding from LabelingManager - this doesn't align well with torchmetrics' standard patterns
Simplicity: Direct sklearn calls are easier to understand, debug, and maintain for our current needs

We've designed our metrics (e.g., RelationWeightedAUC) to be modular, so we can refactor to torchmetrics later if we move to mini-batch validation or distributed training. For now, the added dependency and abstraction overhead isn't justified.
Revisit if: We implement mini-batch validation or need distributed metric aggregation.
